tests/test_anasazi.cc

```diff
@@ -0,0 +1,119 @@
+/*************************************************************************
+ * Copyright (c) 2017-2019 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#define BOOST_TEST_MODULE anasazi
+
+#include "mfmg/dealii/anasazi.templates.hpp"
+
+#include <boost/test/data/test_case.hpp>
+
+#include <cmath>
+#include <cstdio>
+
+#include "lanczos_simpleop.templates.hpp"
+#include "main.cc"
+
+namespace bdata = boost::unit_test::data;
+namespace tt = boost::test_tools;
+
+namespace Anasazi
+{
+
+template <typename VectorType>
+class OperatorTraits<double, mfmg::MultiVector<VectorType>,
+                     mfmg::SimpleOperator<VectorType>>
+{
+  using MultiVectorType = mfmg::MultiVector<VectorType>;
+  using OperatorType = mfmg::SimpleOperator<VectorType>;
+
+public:
+  static void Apply(const OperatorType &op, const MultiVectorType &x,
+                    MultiVectorType &y)
+  {
+    auto n_vectors = x.n_vectors();
+
+    ASSERT(x.size() == y.size(), "");
+    ASSERT(y.n_vectors() == n_vectors, "");
+
+    for (int i = 0; i < n_vectors; i++)
+      op.vmult(*y[i], *x[i]);
+  };
+};
+
+} // namespace Anasazi
+
+BOOST_DATA_TEST_CASE(anasazi,
+                     bdata::make({1, 2}) * bdata::make({1, 2, 3, 5, 10}),
+                     multiplicity, n_distinct_eigenvalues)
+{
+  using namespace mfmg;
+
+  using VectorType = dealii::Vector<double>;
+  using OperatorType = SimpleOperator<VectorType>;
+
+  const int n = 1000;
+  const int n_eigenvectors = n_distinct_eigenvalues * multiplicity;
+
+  OperatorType op(n, multiplicity);
+
+  boost::property_tree::ptree anasazi_params;
+  anasazi_params.put("num_eigenpairs", n_eigenvectors);
+
+  anasazi_params.put("max_iterations", 1000);
+  anasazi_params.put("tolerance", 1e-2);
```

Comment:
Why did you pick this tolerance?

Reply:
This is a tricky one. If I put `1e-4`, 1000 iterations may not be enough for LOBPCG to converge for larger number of eigenvalues.